### PR TITLE
(703c) Assorted prehistoric delights

### PR DESCRIPTION
**Pull request by yndajas**
## Changes in this PR

- Various small fixes, tidy ups and refactors in preparation for the final PR to land 703

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Comment by yndajas** [comment]
[Add programme history index page (UI)](https://trello.com/c/igilNoFW/703-add-programme-history-index-page-ui)

File: server/testutils/factories/courseParticipationOutcome.ts [hunk 1]
```diff
@@ -13,15 +13,15 @@ const outcomeTypes = {
   complete() {
     return {
       detail: faker.lorem.paragraph({ max: 5, min: 0 }),
-      status: 'complete' as CourseParticipationOutcomeStatus,
+      status: 'complete' as CourseParticipationOutcome['status'],
```
Comment (jsrobertson) [inline server/testutils/factories/courseParticipationOutcome.ts:16]: Might be better to do to `complete(): CourseParticipationOutcome` and same for `incomplete()`.  Save having to use `as` and then the other values will be typed too.

File: server/views/referrals/courseParticipations/new.njk [hunk 2]
```diff
@@ -55,4 +55,4 @@
       </form>
     </div>
   </div>
-{% endblock %}
+{% endblock body %}
```
Comment (jsrobertson) [inline server/views/referrals/courseParticipations/new.njk:58]: `{% endblock content %}`?

File: server/testutils/factories/courseParticipationOutcome.ts (hunk 1 above)
Comment (yndajas) [reply server/testutils/factories/courseParticipationOutcome.ts:16]: Good shout

File: server/views/referrals/courseParticipations/new.njk (hunk 2 above)
Comment (yndajas) [reply server/views/referrals/courseParticipations/new.njk:58]: Oops

**Review by jsrobertson — APPROVED** [review]
👍

**Review by Gweaton — APPROVED** [review]

**Pull request merged**